src/integrationTest/java/com/mongodb/hibernate/jdbc/BatchUpdateIntegrationTests.java

```diff
@@ -0,0 +1,278 @@
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 1}}, limit: 0}
+                                                    """),
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 2}}, limit: 0}
+                                                    """),
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 3}}, limit: 0}
+                                                    """));
+                        },
+                        command2 -> {
+                            assertThat(command2.entrySet()).contains(Map.entry("delete", new BsonString("movies")));
+                            assertThat(command2.getArray("deletes").getValues())
+                                    .containsExactly(
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 4}}, limit: 0}
+                                                    """),
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 5}}, limit: 0}
+                                                    """),
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 6}}, limit: 0}
+                                                    """));
+                        },
+                        command3 -> {
+                            assertThat(command3.entrySet()).contains(Map.entry("delete", new BsonString("movies")));
+                            assertThat(command3.getArray("deletes").getValues())
+                                    .containsExactly(
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 7}}, limit: 0}
+                                                    """),
+                                            BsonDocument.parse(
+                                                    """
+                                                    {q: {_id: {$eq: 8}}, limit: 0}
+                                                    """));
+                        },
+                        command4 -> assertThat(command4.getFirstKey()).isEqualTo("commitTransaction"));
+    }
+
+    // TODO-HIBERNATE-19 https://jira.mongodb.org/browse/HIBERNATE-19
```

Comment:
This PR adds a `TODO-HIBERNATE-19` not, but https://github.com/mongodb/mongo-hibernate/pull/59 for https://jira.mongodb.org/browse/HIBERNATE-19 has already been created. Now https://github.com/mongodb/mongo-hibernate/pull/59 (HIBERNATE-19) depends on the current PR (https://github.com/mongodb/mongo-hibernate/pull/35 (HIBERNATE-35)), yet they are being worked on concurrently.

If we proceed as is, it is not impossible that we'll merge https://github.com/mongodb/mongo-hibernate/pull/59 (HIBERNATE-19) first, then the currernt PR (https://github.com/mongodb/mongo-hibernate/pull/35 (HIBERNATE-35)), and the `TODO-HIBERNATE-19` added by the current PR will stay in `main` despite HIBERNATE-19 having been resolved.

What is our plan on making sure the above is not going to happen?

Reply:
seems we can avoid such deadlock by disabling one PR for the time being until the other one got merged?

Reply:
I disabled this PR for the time being so we could focus on #59.

Reply:
I worry that when we create another PR for HIBERNATE-35, it will be based on this one, and the `TODO-HIBERNATE-19` will slip there, despite HIBERNATE-19 having been closed. Can we replace this `TODO-HIBERNATE-19` comment with a `TODO-HIBERNATE-35` comment explaining what needs to be done, and that it should be done after implementing HIBERNATE-19.

Reply:
did it. good point.